src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Entities/CacheableEntity.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace OrchardCore.Entities;
+
+public class CacheableEntity : ICacheableEntity
+{
+    private JObject _properties = new();
+
+    private Dictionary<string, object> _cache { get; } = new Dictionary<string, object>();
+
+    public JObject Properties
+    {
+        get
+        {
+            return _properties;
+        }
+        set
+        {
+            _properties = value ?? new JObject();
+            _cache.Clear();
+        }
+    }
+
+    public void Forget(string key)
+    {
+        AssertNotNull(key);
+
+        _cache.Remove(key);
+    }
+
+    public object Get(string key)
+    {
+        AssertNotNull(key);
+
+        if (_cache.TryGetValue(key, out var value))
+        {
+            return value;
+        }
+
+        return null;
+    }
+
+    public void Set(string key, object value)
+    {
+        AssertNotNull(key);
+
+        if (value == null)
+        {
+            return;
+        }
+
+        _cache[key] = value;
+    }
+
+    private static void AssertNotNull(string key)
```

Comment:
We could remove this

Reply:
done

Reply:
Still we have it :)

Reply:
Yes we have it because we use .net7 locally. But the build in GitHub fails because it runs the code on .net6 and .net7.